### PR TITLE
[RDY] Fix warning: RGB already defined on windows/mingw*

### PR DESCRIPTION
**Pull request by zhaozg**

### CODE REVIEW

File: src/nvim/macros.h [hunk 1]
```diff
@@ -148,7 +148,9 @@
 /// zero in those cases (-Wdiv-by-zero in GCC).
 #define ARRAY_SIZE(arr) ((sizeof(arr)/sizeof((arr)[0])) / ((size_t)(!(sizeof(arr) % sizeof((arr)[0])))))
 
+#ifndef RGB
```
Comment (justinmk) [inline src/nvim/macros.h]: our definition is not necessarily the same, and it is required.

Comment (zhaozg) [reply src/nvim/macros.h]:
```
typedef unsigned long DWORD;
typedef DWORD COLORREF;
#define RGB(r,g,b) ((COLORREF)(((BYTE)(r)|((WORD)((BYTE)(g))<<8))|(((DWORD)(BYTE)(b))<<16)))
```
uint32_t equals unsigned long

Comment (justinmk) [reply src/nvim/macros.h]: That's nice, but rather than depending on that, it could be undefined instead. That's what we do for other systems already.

Comment (zhaozg) [reply src/nvim/macros.h]: I got your means. thanks

Comment (justinmk) [reply src/nvim/macros.h]: I mean we could `#undefine` it here.

**Pull request closed without merging**